com.unity.shadergraph/Editor/GraphUI/DataModel/Searcher/ShaderGraphSearcherDatabaseProvider.cs

```diff
@@ -0,0 +1,47 @@
+using System.Collections.Generic;
+using UnityEditor.GraphToolsFoundation.Overdrive;
+using UnityEditor.GraphToolsFoundation.Searcher;
+using UnityEngine.GraphToolsFoundation.Overdrive;
+
+namespace UnityEditor.ShaderGraph.GraphUI.DataModel
+{
+    public class ShaderGraphSearcherDatabaseProvider : DefaultSearcherDatabaseProvider
+    {
+        public ShaderGraphSearcherDatabaseProvider(Stencil stencil)
+            : base(stencil)
+        {
+        }
+
+        SearcherDatabaseBase CreateNodeDatabaseFromRegistry(IGraphModel graphModel)
```

Comment:
Yeah, if we can eventually query by arbitrary criteria then I'd think we can get a list of everything. Let me know what you find out.

It would be great if we could get some enumerator of nodes to map into `SearcherItems`. One thing I'm curious about is how categories (like Artistic, Channel, and especially nested ones like Input/2D etc.) will be presented to us.

Also not accounted for yet is the dynamic stuff -- blackboard variables, custom interpolators, etc. We can probably generate another `SearcherDatabaseBase` for that and add it to the list a few lines down in `GetGraphElementsSearcherDatabases` when the time comes.